airbyte-scripts/github-filler/main.py

```diff
@@ -0,0 +1,141 @@
+
+    for branch in branches:
+        data = {
+            'title': f'Issue for branch {branch.get("name")}',
+            'head': branch.get('name'),
+            'labels': labels,
+            'milestone': milestone,
+            'assignees': assignees
+        }
+
+        # add issue
+        response = requests.post(url=url, data=json.dumps(data), headers=head)
+        logging.info(f'Response status: {response.status_code}, response body: {"Success" if response.status_code in [200, 201] else response.content}')
+
+        # add issue comment
+        comments_url = response.json().get('comments_url')
+        response = requests.post(url=comments_url, data=json.dumps({'body': f'comment for issues {comments_url}'}), headers=head)
+        logging.info(f'Response status: {response.status_code}, response body: {"Success" if response.status_code in [200, 201] else response.content}')
+
+
+def create_release(rep: str, head: dict):
+    url = f"https://api.github.com/repos/{rep}/releases"
+    for i in range(10):
+        data = {
+            'tag_name': f'dev-0.{i}',
+            'name': '{i} global release'
+        }
+        response = requests.post(url=url, data=json.dumps(data), headers=head)
+        logging.info(f'Response status: {response.status_code}, response body: {"Success" if response.status_code in [200, 201] else response.content}')
+
+
+@logger
+def star_a_repository(rep: str, head: dict):
+    url = f"https://api.github.com/user/starred/{rep}"
+    return requests.put(url=url, headers=head)
+
+
+def create_project(rep: str, head: dict):
+    url = f"https://api.github.com/repos/{rep}/projects"
+    for name in ['project_1', 'project_2', 'project_3']:
+        response = requests.post(url=url, data=json.dumps({'name': name}), headers=head)
+        logging.info(f'Response status: {response.status_code}, response body: {"Success" if response.status_code in [200, 201] else response.content}')
+    
+
+def create_milestone(rep: str, head: dict):
+    url = f"https://api.github.com/repos/{rep}/milestones"
+    for title in ['main', 'test', 'feature']:
+        data = {
+            'title': title
+        }
+        response = requests.post(url=url, data=json.dumps(data), headers=head)
+        logging.info(f'Response status: {response.status_code}, response body: {"Success" if response.status_code in [200, 201] else response.content}')
+    
+
+if __name__ == "__main__":
+
+    api_token = sys.argv[1]
+    repository = sys.argv[2]
+
+    headers = get_headers(api_token)
+    branches = get_all_branches(repository, headers)
+    
+    pull_requests(repository, branches, headers)
+    
+
+    create_milestone(repository, headers)
+    labels = ['important', 'bug', 'critical']
+    milestone = 1
+    assignees = []
+
+    create_release(repository, headers)
+    star_a_repository(repository, headers)
+    create_project(repository, headers)
+    add_issues_with_comments(repository, branches, headers, labels, milestone, assignees)
```

Comment:
since there are 8 function related to one instance (github) is not a better to put them all inside class ?
Also this can support the following: 
```def __init__(self)`__: 
    self.session = requests.Session()
    self.session.headers.update(self.get_headers(token))
```
and further we can forget to use headers everywhere, just sending with `seff.session.post(...)`

E.g we have one class as container, methods inside who shares and uses the same session. Also the same with repository, could be used as `self.repository` and not be passed to every function

Reply:
also a lot of usages the url with the same base `https://api.github.com`
Move this to file constant or class consant/property (like BASE_URL, url_base etc)